pkg/storage/chunk/predicate.go

```diff
@@ -0,0 +1,16 @@
+package chunk
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/logql/syntax"
+)
+
+type Predicate struct {
```

Comment:
Maybe we can come up with a better name.

Reply:
`Filter` or `Selector` are the only alternatives I find fitting.